velox/functions/remote/client/ThriftClient.cpp

```diff
@@ -15,13 +15,16 @@
  */
 
 #include "velox/functions/remote/client/ThriftClient.h"
+#include <thrift/lib/cpp2/async/HeaderClientChannel.h>
 
 namespace facebook::velox::functions {
 
 std::unique_ptr<RemoteFunctionClient> getThriftClient(
     folly::SocketAddress location,
     folly::EventBase* eventBase) {
-  return newHeaderClient<RemoteFunctionClient>(eventBase, location);
+  auto sock = folly::AsyncSocket::newSocket(eventBase, location);
+  return std::make_unique<RemoteFunctionClient>(
```

Comment:
One slightly unrelated question : We have been doing a lot of work on ensuring backwards compatibility of function signatures for udfs in Velox - How can we ensure the same for remote udfs ? Is the registry for remote udfs separate from the non remote ones.

Reply:
Good question. Remote functions are registered in the same registry, so whatever you're doing for scalar function should directly apply here.

Reply:
Also curious, if they share the same registry how will they work with our fuzzers ? Do we have any plans to filter them out ?

Reply:
They should work out of the box with the fuzzers, as long as you register any before your fuzzer run (which we don't, at least for now).